docs/training-debug/index.txt

```diff
@@ -1,343 +1,395 @@
-Once you are on the cluster, testing is identical to Step 3, with the important caveat that the
-model definition argument to ``det experiment create`` (the second positional argument) should
-always be ``/run/determined/workdir`` (or ``.`` if you have not changed the working directory from
-when you first connected to the cluster).
+   If you are using SSH to interact with your model, enter:
 
-This is because the ``--context`` you passed when creating the shell or notebook will be copied to
-``/run/determined/workdir`` inside the container, exactly the same as the model definition argument
-to ``det experiment create`` would.
+   .. code:: bash
 
-**How to diagnose failures:** Failure diagnosis is a combination of the failure diagnosis for Step 3
-and Step 4.
+      det shell start --context my_model_dir
+      # Your terminal should automatically connect to the shell.
 
-*********************
- Higher-level Issues
-*********************
+   When interacting with the shell or notebook, testing is the same as :ref:`Step 2 <step2>`.
 
-7. Does cluster test mode work with ``slots_per_trial`` set to ``1``?
-=====================================================================
+#. Perform failure diagnostics:
 
-This step is conceptually similar to Step 6, except instead of launching the command from an
-interactive environment, we will submit it to the cluster and let Determined manage everything.
+   Combine the failure diagnosis steps used in :ref:`Step 2 <step2>` and :ref:`Step 4 <step4>`.
 
-**How to test:** If you had to make any customizations to your command environment while testing
-Steps 3, 4, or 5, make sure that you have made the same customizations in your experiment config.
-Then also confirm that your experiment config either does not specify ``resources.slots_per_trial``
-at all, or that it is set to 1, like:
+.. _step6:
 
-.. code:: yaml
+*******************************************************************
+ Step 6 - Verify that local test mode works in a notebook or shell
+*******************************************************************
 
-   resources:
-     slots_per_trial: 1
+This step is the same as :ref:`Step 3 <step3>`, except the original code runs on the Determined
+cluster instead of locally.
 
-Then create an experiment with the ``--test`` flag (but not the ``--local`` flag):
+#. Launch a notebook or shell as described in :ref:`Step 4 <step4>`.
 
-.. code:: bash
+   On the cluster, testing is the same as :ref:`Step 3 <step3>`, except that the second, model
+   definition argument of the ``det experiment create`` command should be
+   ``/run/determined/workdir``, or ``.``, the current directory if you have not changed the working
+   directory after connecting to the cluster. This is because the ``--context`` specified when
+   creating the shell or notebook is copied to the ``/run/determined/workdir`` directory inside the
+   container, the same as the model definition argument is copied to ``det experiment create``.
```

Comment:
The commas here seem awkward.

> except that the second, model definition argument of the ``det experiment create`` command

Here "second, model definition argument" seems grammatically correct ("second" and "model definition" are two adjectives that both apply) but it feels less clear than just saying the "model definition argument".  I say this because from a CLI perspective, both `experiment` and `create` are technically arguments to the `det` command, but functionally `experiment create` represents a subcommand and a subsubcommand, with arguments after it, so numbering arguments is not crystal clear in this situation.

> should be ``/run/determined/workdir``, or ``.``

No comma after workdir: "should be X or Y".

> or ``.``, the current directory if you

"the current directory" is an appositive, and so needs a comma after it.  But in this case, I would guess it's widely enough known that `.` represents the current directory, and I would just remove the `, the current directory,` appositive entirely.

Reply:
Fixed. All good suggestions.